### DISCUSSION TITLE
Compiling Java 17 - JDK 17 mvn-runtime requirement expected?

### DISCUSSION THREAD

**Discussion by aumann** [eclipse-tycho/tycho#938] [General]
Hey everyone,

I'm asking mostly out of curiosity: is it expected that Java 17 code only compiles when maven/tycho is running on a JDK 17?

My intuition would have been that `BREE`,  Manifest `JavaSE-17` and setting the execution environment in the target platform to `JavaSE-17` is enough. However, as soon as we use actual additions from the 17 runtime library (like `List.toStream()`) the compiler starts complaining. The compile errors vanish as soon as tycho itself runs on a JDK 17.

Maybe my expectations are just wrong? Otherwise, I'd open an issue.

P.S. I've tested Tycho versions 2.6.0 and 2.7.1

**Comment by akurtakov** [comment]
There are 2 ways for this to work:
* Run on Java 17 - simplest solution
* Use toolchains - aka a file similar to https://github.com/eclipse/tycho/blob/master/toolchains-gh.xml pointing to exact location of Java 17 you have is placed in ~/.m2/toolchains.xml and tycho-compiler-plugin is configured to ```<useJDK>BREE</useJDK>```

**Reply by aumann** [reply]
Thank you!
We have a toolchains file, and tycho picks up the JDK according to the logs. So this should indeed work? That is, if tycho reports that it uses the JDK 17 for compiling, it should also compile running on Java 11?

Then I'll try to minimize the issue and open a bug report as soon as I find some spare time.

**Reply by akurtakov** [reply]
Please do, without reproducer it's guess work.

**Comment by mickaelistria** [comment]
> That is, if tycho reports that it uses the JDK 17 for compiling, it should also compile running on Java 11

Yes.
